gulp/styles.js

```diff
@@ -53,56 +71,49 @@ export default function styles(done) {
 					'nesting-rules': true
 				}
 			)
-		})
+		}),
+		cssnano()
 	];
 
-	// Only minify if we aren't building for
-	// production and debug is not enabled
-	if( ! config.dev.debug.styles && ! isProd ) {
-		postcssPlugins.push(cssnano());
+	// Skip minifying files if we aren't building for
+	// production and debug is enabled
+	if( config.dev.debug.styles && ! isProd ) {
+		postcssPlugins.pop();
 	}
 
 	// Report messages from other postcss plugins
 	postcssPlugins.push(
 		reporter({ clearReportedMessages: true })
 	);
 
-	const beforeReplacement = [
-		src( paths.styles.src, {sourcemaps: !isProd} ),
-		logError('CSS'),
-		gulpPlugins.newer({
-			dest: paths.styles.dest,
-			extra: [paths.config.themeConfig]
-		}),
-		gulpPlugins.phpcs({
-			bin: `${rootPath}/vendor/bin/phpcs`,
-			standard: 'WordPress',
-			warningSeverity: 0
-		}),
-		// Log all problems that were found.
-		gulpPlugins.phpcs.reporter('log'),
-	];
-
-	const afterReplacement = [
+	// Return a single stream containing all the
+	// after replacement functionality
+	return pipeline.obj([
 		gulpPlugins.postcss(postcssPlugins),
 		gulpPlugins.rename({
 			suffix: '.min'
 		}),
 		server.stream({match: "**/*.css"}),
-		dest(paths.styles.dest, {sourcemaps: !isProd}),
-	];
+	]);
+}
 
-	pump(
-		[].concat(
-			beforeReplacement,
-			// Only do string replacements when building for production
-			gulpPlugins.if(
-				isProd,
-				getStringReplacementTasks(),
-				[]
-			),
-			afterReplacement
+/**
+* CSS via PostCSS + CSSNext (includes Autoprefixer by default).
+*/
+export default function styles(done) {
+
+	// Convert the array of string replacement tasks into a single stream object
+	const stringReplacementTasks = pipeline.obj( getStringReplacementTasks() );
```

Comment:
I'd probably move this into the `getStringReplacementTasks()` method.

Reply:
Yeah, that’s a good suggestion. Right now some of the other tasks expect an array.

I’ll update the rest of the tasks to use pipeline then update the return type of getStringReplacementTasks